internal/controller/serving/inferencegraph_controller.go

```diff
@@ -166,12 +166,23 @@ func (r *InferenceGraphReconciler) getExistingAuthConfig(ctx context.Context, ig
 }
 
 // SetupWithManager sets up the controller with the Manager.
-func (r *InferenceGraphReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	return ctrl.NewControllerManagedBy(mgr).
+func (r *InferenceGraphReconciler) SetupWithManager(mgr ctrl.Manager, isServingMode bool) error {
+	builder := ctrl.NewControllerManagedBy(mgr).
 		For(&servingv1alpha1.InferenceGraph{}).
-		Owns(&authorinov1beta2.AuthConfig{}).
-		Named("serving-inferencegraph").
-		Complete(r)
+		Named("serving-inferencegraph")
+	// dynamic add authconfig to watchlist based on serving mode + if authconfig crd is available
+	authConfigCrdAvailable, authCrdErr := utils.IsCrdAvailable(
+		mgr.GetConfig(),
+		authorinov1beta2.GroupVersion.String(),
+		"AuthConfig")
+	if authCrdErr != nil {
+		return fmt.Errorf("failed to check if  AuthConfig CRD in the cluster: %w", authCrdErr)
+	}
+	if authConfigCrdAvailable && isServingMode {
```

Comment:
I think a fast-fail way would be better. so if isServingMode is false then it should return right away.

Reply:
thanks, did some updated